### PR TITLE
Changes made to the User Interface of Quiz Engine

### PR DESCRIPTION
**Pull request by aryajain14**
## Summary

- Added scrolling feature on the splash screen
- In the question header,
   - Question Index and Type is aligned to the left side
   - Subject name and section are aligned to the right side
- Added a panel which shows Test name and Id on the left and right respectively.
- Testing was done for the above newly added features (5 tests each on vue-test-utils and cypress)
- ADR Link: https://www.notion.so/avantifellows/ADR-Adding-a-Header-to-QE-b9c829039c6c4cce9bdd2ab480a2caa4

Testing Link 1: https://staging-quiz.avantifellows.org/quiz/647ddb837c11b58610a8d081?userId=666666&apiKey=6qOO8UdF1EGxLgzwIbQN
(This testing link contains a longer test-name to see truncation property)

Testing Link 2: https://staging-quiz.avantifellows.org/quiz/647ddc0da36f9c0c3ded6e52?userId=666666&apiKey=6qOO8UdF1EGxLgzwIbQN
(This testing link contains a shorter test-name)

Testing link 3: https://staging-quiz.avantifellows.org/quiz/647ddc8e7c11b58610a8d0e6?userId=666666&apiKey=6qOO8UdF1EGxLgzwIbQN 
(For OMR type quiz)

## Test Plan

- [x] Test Responsiveness
  - [x] Laptop (1200px)
  - [x] Tablet (760px)
  - [x] Phone (320px)
- [x] Cross-Browser Testing
  - [x] Chrome
  - [x] Firefox
- [ ] ~Local Language Support~
- [x] Hygiene and Housekeeping
  - [x] Self-review
  - [x] Comments have been added appropriately
  - [ ] ~Check for bundle size [here](https://bundlephobia.com/) if adding a package~
  - [ ] ~Added relevant details like Labels/Projects/Milestones etc.~
  - [ ] ~If adding or removing any environment variable, update `docs/ENV.md`, `.env.example` and the Github workflows.~
- [x] Testing
  - [x] Wrote tests
  - [x] Tested locally
  - [x] Tested on staging
  - [x] Tested on an actual physical phone
  - [x] Tested on production
- [ ] ~Lighthouse Checks~
  - [ ] ~Images have `alt` attributes~
  - [ ] ~Any `` tags have `width` and `height` specified~
  - [ ] ~Any `target="_blank"` links have `rel="noopener"`~
  - [ ] ~Only SVGs are used as images. If PNGs are used, their size has been optimised.~
  - [ ] ~Any SVG buttons without text have their `aria-label` attributes set~

### CODE REVIEW

File: src/components/Questions/QuestionModal.vue [hunk 1]
```diff
@@ -15,6 +15,18 @@
     <div
       class="flex flex-col grow bg-white w-full justify-between overflow-hidden"
     >
+    <div class="grid grid-rows-1 grid-flow-col gap-2 ">
+          <div class="row-span-1 text-left h-10 mx-4 py-3 text-xl text-base sm:text-xs md:text-base lg:text-lg xl:text-xl 2xl:text-2xl" style = "@media (max-width: 870px){font-size: 0.75rem; line-height: 1rem;}">
```
Comment (deepansh96) [inline src/components/Questions/QuestionModal.vue:19]: why are there tailwind classes along with an inline style attribute? 
`style = "@media (max-width: 870px){font-size: 0.75rem; line-height: 1rem;}">`
You can rewrite this thing using tailwind itself. You can extend the tailwind config and put a custom breakpoint. Take a look at the `tailwind.config.js` file in this project and see how we've added custom breakpoints. Similarly you can add the 870px waala breakpoint and then set the font size and line height

File: src/components/Splash.vue [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <template>
-  <div class="flex flex-col justify-center h-full" @wheel="preventScrolling">
+  <!-- <div class="flex flex-col justify-center h-full" @wheel="preventScrolling"> -->
```
Comment (deepansh96) [inline src/components/Splash.vue:2]: comment

File: src/components/Splash.vue [hunk 3]
```diff
@@ -160,14 +157,25 @@ export default defineComponent({
...
-      preventScrolling,
       start,
     };
   },
   emits: ["start"],
 });
 </script>
 
+methods: {
+  scrollContent(event) {
```
Comment (deepansh96) [inline src/components/Splash.vue:168]: what does this method do? can you add a comment above this method explaining this?

File: src/views/Player.vue [hunk 4]
```diff
@@ -59,6 +59,8 @@
         :qsetIndexLimits="currentQsetIndexLimits"
         :quizTimeLimit="quizTimeLimit"
         :timeRemaining="timeRemaining"
+        :userId="userId"
+        :title="title"
```
Comment (deepansh96) [inline src/views/Player.vue:63]: this doesn't seem like it is covered in tests, can you check?

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: src/components/Scorecard.vue [hunk 5]
```diff
@@ -199,7 +199,7 @@ export default defineComponent({
       backButtonClass:
         "bg-[#F78000] hover:bg-primary-hover bp-500:w-40 px-6 py-3 bp-500:p-4 bp-500:px-10 sm:p-6 rounded-2xl md:rounded-xl shadow-xl disabled:opacity-50 disabled:pointer-events-none invisible",
       shareButtonClass:
-        "bg-[#10B981] hover:bg-green-600 bp-500:w-40 px-6 py-3 bp-500:p-4 bp-500:px-10 sm:p-6 rounded-2xl md:rounded-xl shadow-xl disabled:opacity-50 disabled:pointer-events-none",
+        "flex justify-center bg-[#10B981] hover:bg-green-600 bp-500:w-40 px-6 py-3 bp-500:p-4 bp-500:px-10 sm:p-6 rounded-2xl md:rounded-xl shadow-xl disabled:opacity-50 disabled:pointer-events-none",
```
Comment (sudheshna-donthineni) [inline src/components/Scorecard.vue:202]: why don't you add this colors to the theme?

File: tests/e2e/specs/renders-omr-assessment.js [hunk 6]
```diff
@@ -343,85 +374,85 @@ describe("Player for OMR quizzes", () => {
...
-
-        cy.get("@omr-modal")
-          .get('[data-test="OmrItem-0"]')
-          .within(() => {
-            cy.get('[data-test="optionSelector-0"]').should("not.be.checked");
-          });
-      });
-    });
-  });
+  // describe("Resume Session", () => {
```
Comment (sudheshna-donthineni) [inline tests/e2e/specs/renders-omr-assessment.js:377]: why not delete the commented code?

File: src/components/Questions/Body.vue [hunk 7]
```diff
@@ -130,7 +132,7 @@
           <!-- input area for the answer -->
           <Textarea
             v-model:value="numericalAnswer"
-            class="px-2 w-full"
+            class="px-2 w-full mb-20"
```
Comment (suryabulusu) [inline src/components/Questions/Body.vue:135]: remove this

File: src/components/Questions/Palette/QuestionPalette.vue [hunk 8]
```diff
@@ -29,7 +29,7 @@
         <div class="grid grid-cols-5 bp-500:grid-cols-6 lg:grid-cols-7 xl:grid-cols-8 mt-4 space-y-4">
           <PaletteItem
             v-for="(questionState, qindex) in questionSetState.paletteItems"
-            class="hover:cursor-pointer"
+            class="hover:cursor-pointer mb-12"
```
Comment (suryabulusu) [inline src/components/Questions/Palette/QuestionPalette.vue:32]: u can remove this -- check once

**Review by suryabulusu — APPROVED** [review]
looks good to me! good job!

**Pull request merged**